xtask/src/tidy.rs

```diff
@@ -334,6 +335,36 @@ fn check_dbg(path: &Path, text: &str) {
     }
 }
 
+fn check_test_attrs(path: &Path, text: &str) {
+    let ignore_rule =
+        "https://github.com/rust-analyzer/rust-analyzer/blob/master/docs/dev/style.md#ignore";
+    let need_ignore: &[&str] = &[
+        // Special case to run `#[ignore]` tests
+        "ide/src/runnables.rs",
+        // A legit test which needs to be ignored, as it takes too long to run
+        // :(
+        "hir_def/src/nameres/collector.rs",
+        // Obviously needs ignore.
+        "ide_assists/src/handlers/toggle_ignore.rs",
+        // See above.
+        "ide_assists/src/tests/generated.rs",
+    ];
+    if text.contains("#[ignore") && !need_ignore.iter().any(|p| path.ends_with(p)) {
+        panic!("\ndon't `#[ignore]` tests, see:\n\n    {}\n\n   {}\n", ignore_rule, path.display(),)
+    }
+
+    let panic_rule =
+        "https://github.com/rust-analyzer/rust-analyzer/blob/master/docs/dev/style.md#panic";
```

Comment:
This is supposed to link to https://github.com/rust-analyzer/rust-analyzer/blob/master/docs/dev/style.md#should_panic right? not #panic